From d18e3d64be82098d826bf7b323b09b19d4c7a20f Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:12:51 -0400
Subject: [powerpc] Document sysfs DSCR interface

Message-id: <1410545655-205645-543-git-send-email-dzickus@redhat.com>
Patchwork-id: 94436
O-Subject: [RHEL7 PATCH 542/626] powerpc: Document sysfs DSCR interface
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 26c88f930141f0d9e7673af2332ada67b7e8865a
Author: Sam bobroff <sam.bobroff@au1.ibm.com>
Date:   Wed May 21 16:32:39 2014 +1000

    powerpc: Document sysfs DSCR interface

    Add some documentation about ...

    /sys/devices/system/cpu/dscr_default
    /sys/devices/system/cpu/cpuN/dscr

    ... to Documentation/ABI/stable.

    Signed-off-by: Sam Bobroff <sam.bobroff@au1.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---
 create mode 100644 Documentation/ABI/stable/sysfs-devices-system-cpu

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/ABI/stable/sysfs-devices-system-cpu b/Documentation/ABI/stable/sysfs-devices-system-cpu
new file mode 100644
index 0000000..33c133e
--- /dev/null
+++ b/Documentation/ABI/stable/sysfs-devices-system-cpu
@@ -0,0 +1,25 @@
+What:   /sys/devices/system/cpu/dscr_default
+Date:  13-May-2014
+KernelVersion: v3.15.0
+Contact:
+Description: Writes are equivalent to writing to
+  /sys/devices/system/cpu/cpuN/dscr on all CPUs.
+  Reads return the last written value or 0.
+  This value is not a global default: it is a way to set
+  all per-CPU defaults at the same time.
+Values:  64 bit unsigned integer (bit field)
+
+What:   /sys/devices/system/cpu/cpu[0-9]+/dscr
+Date:  13-May-2014
+KernelVersion: v3.15.0
+Contact:
+Description: Default value for the Data Stream Control Register (DSCR) on
+  a CPU.
+  This default value is used when the kernel is executing and
+  for any process that has not set the DSCR itself.
+  If a process ever sets the DSCR (via direct access to the
+  SPR) that value will be persisted for that process and used
+  on any CPU where it executes (overriding the value described
+  here).
+  If set by a process it will be inherited by child processes.
+Values:  64 bit unsigned integer (bit field)
-- 
1.7.1